### PR TITLE
fix: add created date column and implement sorting by created date

### LINKED ISSUE #304: Add Created Date Column and Implement Sorting by Created Date
## Description

 When editing an item's name, the display order changes unexpectedly. To resolve this, we need to add a "createdDate" column to our database table and implement sorting by this column in our Prisma query. This will ensure items are consistently displayed in the order they were created.

### PR DESCRIPTION
**Pull request by KaiKoide**
## Overview

Resolve the unexpected behavior where renaming an item changes its display order. This issue was intended to be resolved by adding createdAt columns and using the orderBy option. However, to handle cases where items have the same createdAt value, I also included the item ID in the orderBy option.

## Changes

- Add `createdAt` column, with a default value set to the time, to Tenant, Category, Task, and ShareHouse table.
- Add `orderBy` option when fetching data from DB

## Screenshots or videos

**Before editing name**
image

**After editing name**
image

## Assignee Checklist:

- [ ] The base branch is correct (no accidental merges)
- [ ] The branch name follows our branch naming rules
- [ ] The PR title follows our PR title rules
- [ ] My code follows our coding style
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings or errors in the console

## Reviewer Checklist:

- [x] Code readability and simplicity
- [x] Follows best practices and coding standards
- [x] Understandable and maintainable code

### CODE REVIEW

File: src/app/api/[[...route]]/routes/sharehouse.route.ts
Comment (akiomatic) [inline src/app/api/[[...route]]/routes/sharehouse.route.ts:1]: Could you add sorting by created date to the share houses?

File: src/app/api/[[...route]]/routes/sharehouse.route.ts [hunk 1]
```diff
@@ -16,6 +16,31 @@ import { InitialAssignedData } from '@/services/InitialAssignedData';
...
+        id: string;
+        name: string;
+        email: string;
+        createdAt: Date;
+      } | null;
+    }[];
+    rotationCycle: number;
+  } | null;
+}
+
```
Comment (akiomatic) [inline src/app/api/[[...route]]/routes/sharehouse.route.ts]: Could you move this to `/src/types/server.ts`?
Also, could you change the writing style to something like as following: https://github.com/Tascurator/tascurator-frontend/blob/5273ed95d352a327f50bbff950b212b75db3699a/src/types/server.ts#L115

**Review by akiomatic — CHANGES_REQUESTED** [review]
Hi. Thank you. Could you check my comments?

Also, I realized that you would need to apply the sorting to the route that returns the list of the categories with assigned tenants for the next rotation cycle. Therefore, could you do this after PR #296 gets merged?

File: src/app/api/[[...route]]/routes/sharehouse.route.ts
Comment (KaiKoide) [reply src/app/api/[[...route]]/routes/sharehouse.route.ts:1]: I added that.

File: src/app/api/[[...route]]/routes/sharehouse.route.ts (hunk 1 above)
Comment (KaiKoide) [reply src/app/api/[[...route]]/routes/sharehouse.route.ts]: I fixed the tyoe.

**Review by akiomatic — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**